js/src/sdk/utils/error.ts

```diff
@@ -1,179 +1,125 @@
-                }
-            }
-            else{
-                const description = axiosError.response.data.message || axiosError.response.data.error || axiosError.message;
-                errorDetails = {
-                    message: axiosError.message,
-                    description: description,
-                    possibleFix: "Please check the network connection, request parameters, and ensure the API endpoint is correct."
-                }
+        if (!isAxiosError) {
+             this.throwCustomError(SDK_ERROR_CODES.COMMON.UNKNOWN, {
+                message: error.message,
+                description: "",
+                possibleFix: "Please check error message and stack trace",
+                originalError: error,
+                metadata: {}
+            });
+        } else {
+            const isResponseNotPresent = !('response' in error);    
+            if(isResponseNotPresent) {
+                this.handleNonResponseAxiosError(error as AxiosError);
             }
+            this.throwAPIError(error as AxiosError);
         }
+    }
+    private static handleNonResponseAxiosError(error: AxiosError) {
+        const fullUrl = (error.config?.baseURL || "") + (error.config?.url || "");
+        const metadata = generateMetadataFromAxiosError(error);
 
+        if (error.code === "ECONNREFUSED") {
+            throw new ComposioError(
+                SDK_ERROR_CODES.COMMON.BASE_URL_NOT_REACHABLE,
+                `ECONNREFUSED for ${fullUrl}`,
+                "",
+                "Make sure:\n1. The base URL is correct and is accessible\n2. Your network connection is stable\n3. There are no firewall rules blocking the connection",
+                metadata,
+                error
+            );
+        }
 
-        let axiosDataMessage = axiosError.response?.data?.message || axiosError.message;
-        const status = axiosError.response?.status || axiosError.status || axiosError.code || 'unknown';
-        const request_id = axiosError.response?.headers?.["x-request-id"];
-        const urlAndStatus = axiosError.config?.url ? ` got 📊 ${status} response from URL🔗: ${axiosError.config.url}, request_id: ${request_id}` : '';
+        if (error.code === "ECONNABORTED" || error.code === "ETIMEDOUT") {
+            throw new ComposioError(
+                SDK_ERROR_CODES.COMMON.REQUEST_TIMEOUT,
+                `ECONNABORTED for ${fullUrl}`,
+                `Request to ${fullUrl} timed out after the configured timeout period. This could be due to slow network conditions, server performance issues, or the request being too large. Error code: ECONNABORTED`,
+                "Try:\n1. Checking your network speed and stability\n2. Increasing the request timeout setting if needed\n3. Breaking up large requests into smaller chunks\n4. Retrying the request when network conditions improve\n5. Contact tech@composio.dev if the issue persists",
+                metadata,
+                error
+            );
+        }
```

Comment:
In the function handleNonResponseAxiosError, when the error code is "ETIMEDOUT", the error message and description incorrectly reference "ECONNABORTED" instead of "ETIMEDOUT". This misrepresentation can mislead users regarding the actual nature of the error encountered.